R/qenv-extract.R

```diff
@@ -0,0 +1,50 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `qenv` environment and limit the code to the necessary needed to build limited objects.
+#' `...` passes parameters to further methods.
+#'
+#' @param x (`qenv`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' q <- qenv()
+#' q <- eval_code(q, "a <- 1;b<-2")
+#' q["a"]
+#' q[c("a", "b")]
+#'
+#' @rdname qenv
+#'
+#' @export
+`[.qenv` <- function(x, names, ...) {
+  checkmate::assert_class(names, "character")
+  possible_names <- ls(get_env(x), all.names = TRUE)
+  names_warn <- setdiff(names, possible_names)
+  names <- intersect(names, possible_names)
+  if (!length(names)) {
```

Comment:
what if?

```r
q <- within(qenv(), {
	a <- iris
	b <- a
	plot(a)
	rm(a)
}) |> dev_suppress()

q["a"]


```

Reply:
Sure, happy to chat whenever you can! Ready when you are